Subject: fbdev: tag by scantype in sysfs
Modify the sysfs description of a video mode such that modes are tagged with
their scan type, (p)rogessive, (i)nterlaced, (d)ouble scan. For example,
U:1920x1080i-50. This is useful to disambiguate some of the CEA-861 video
format timings (especially those for EDTV).

Signed-off-by: Daniel R Thompson <daniel.thompson@st.com>
Index: linux-2.6.17.14_stm22_0038/drivers/video/fbsysfs.c
===================================================================
--- linux-2.6.17.14_stm22_0038.orig/drivers/video/fbsysfs.c	2006-06-18 02:49:35.000000000 +0100
+++ linux-2.6.17.14_stm22_0038/drivers/video/fbsysfs.c	2007-03-02 10:59:07.000000000 +0000
@@ -95,13 +95,22 @@
 		       const struct fb_videomode *mode)
 {
 	char m = 'U';
+	char v = 'p';
+
 	if (mode->flag & FB_MODE_IS_DETAILED)
 		m = 'D';
 	if (mode->flag & FB_MODE_IS_VESA)
 		m = 'V';
 	if (mode->flag & FB_MODE_IS_STANDARD)
 		m = 'S';
-	return snprintf(&buf[offset], PAGE_SIZE - offset, "%c:%dx%d-%d\n", m, mode->xres, mode->yres, mode->refresh);
+
+	if (mode->vmode & FB_VMODE_INTERLACED)
+		v = 'i';
+	if (mode->vmode & FB_VMODE_DOUBLE)
+		v = 'd';
+
+	return snprintf(&buf[offset], PAGE_SIZE - offset, "%c:%dx%d%c-%d\n",
+	                m, mode->xres, mode->yres, v, mode->refresh);
 }
 
 static ssize_t store_mode(struct class_device *class_device, const char * buf,
